tests/e2e/toDataURL.test.js

```diff
@@ -0,0 +1,235 @@
+    "RBVO3BQW7kQAwEwSxC//9y7h55akCQxvYQjIj/scYo1ijFGqVYoxRrlGKNUqxRijVKsUYp",
+    "1ijFGqVYoxRrlGKNUqxRijXKxUNJ+EkqdyShU+mS0Kl0SfhJKk8Ua5RijVKsUS5epvKmJD",
+    "yh8iaVNyXhTcUapVijFGuUiw9Lwh0qdyShU+mS0Kl0Kk8k4Q6VTyrWKMUapVijXHw5lROV",
+    "kyR0Kt+sWKMUa5RijXIxTBI6lS4JkxVrlGKNUqxRLj5M5Tcl4UTlCZW/pFijFGuUYo1y8b",
+    "Ik/KQkdCpdEjqVLgmdykkS/rJijVKsUYo1ysVDKt9M5UTlmxRrlGKNUqxRLh5Kwh0qXRJ+",
+    "UxLuULkjCZ3KJxVrlGKNUqxRLh5S6ZLQqXRJ6FS6JHQqXRKeSEKn0iWhUzlJwolKl4QTlS",
+    "eKNUqxRinWKBe/LAmdSpeETuUkCZ1Kl4QTlS4Jd6h0SehUuiS8qVijFGuUYo1y8WFJ6FS6",
+    "JJyofFISOpVOpUtCp3KicqLypmKNUqxRijXKxYep3JGEE5UuCZ3KHSp3qHRJ6FR+U7FGKd",
+    "YoxRol/scXS8ITKidJeEKlS8KJyhPFGqVYoxRrlIuHkvCTVE5U7kjCicpJEk6S8JOKNUqx",
+    "RinWKBcvU3lTEu5IwolKp/KEyh1J6FTeVKxRijVKsUa5+LAk3KHyJpWTJHQqdyShU/lNxR",
+    "qlWKMUa5SLL6fSJaFLwhNJeCIJP6lYoxRrlGKNcvHlknCicpKEE5UuCSdJOFHpktCpPFGs",
+    "UYo1SrFGufgwlZ+k0iWhU+lUnlDpktCpdEnoVN5UrFGKNUqxRrl4WRL+EpU7ktCpdCpdEj",
+    "qVO5LQqTxRrFGKNUqxRon/scYo1ijFGqVYoxRrlGKNUqxRijVKsUYp1ijFGqVYoxRrlGKN",
+    "UqxRijXKP0OHEepgrecVAAAAAElFTkSuQmCC",
+  ].join("");
+
+  it("should throw if no arguments are provided", async () => {
+    await expect(() => toDataURL()).toThrow("String required as first argument");
+  });
+
+  it("should generate a valid URL using promise with error correction level L", async () => {
+    await expect(toDataURL("i am a pony!", {
+      ...defaultOptions,
+      maskPattern: 0,
+      errorCorrectionLevel: "L",
+      type: "image/png"
+    })).resolves.toBe(expectedDataURL);
+  });
+
+  it("should trigger error with version 1 and error correction level H", async () => {
+    await toDataURL(
+      "i am a pony!",
+      { 
+        ...defaultOptions,
+        version: 1, 
+        errorCorrectionLevel: "H", 
+        type: "image/png" 
+      },
+      (err, url) => {
+        expect(err).not.toBeNull();
+        expect(url).toBeUndefined();
+      }
+    );
+  });
+
+  it("should return a promise", async () => {
+    const result = toDataURL("i am a pony!");
+    expect(typeof result.then).toBe("function");
+  });
+
+  it("should generate a valid URL using promise with error correction level L", async () => {
+    try {
+      const url = await toDataURL("i am a pony!", { 
+        ...defaultOptions,
+        errorCorrectionLevel: "L", 
+        type: "image/png" 
+      });
+      expect(url).toBe(expectedDataURL);
+    } catch (err) {
+      // Fail the test if we get an unexpected error
+      expect(err).toBeNull();
+    }
+  });
+
+  it("should reject promise with error correction level H", async () => {
+    try {
+      await toDataURL("i am a pony!", {
+        ...defaultOptions,
+        version: 1,
+        errorCorrectionLevel: "H",
+        type: "image/png",
+      });
+    } catch (err) {
+      expect(err).not.toBeNull();
+    }
+  });
+});
+
+describe("Canvas toDataURL - image/png", () => {
+  const expectedDataURL = [
```

Comment:
Why this is an array? Should we use an array of just a string?

Reply:
there is .join("")  at the end of the array 

it's created by joining multiple string segments. Each element in the array represents a portion of a long base64-encoded PNG image. The .join("") method concatenates these elements into a single string, producing the complete base64 data URL.

Reply:
@zelalemyetsedaw Ok, so is not better to have the entire string instead of join the array of strings?

Reply:
Yes, when it is in the form of an array of strings, it is better for readability, maintainability, and line length limitations imposed by linting